astropy/io/ascii/src/tokenizer.c

```diff
@@ -601,10 +601,15 @@ double str_to_double(tokenizer_t *self, char *str)
     {
         val = strtod(str, &tmp);
 
-        if (tmp == str || *tmp != 0)
+        if (tmp == str || *tmp != 0) {
+		    if (0 == strncmp(str, "nan", 3)) {
+				return NAN;
+		    }
```

Comment:
Turns out I had vim configured to use tabs in C files already using tabs, and spaces otherwise.  Looks like there were several lines in that file that are using tabs already so I'll fix them too.